client/X11/xf_window.c

```diff
@@ -203,10 +206,16 @@ void xf_SetWindowFullscreen(xfContext* xfc, xfWindow* window, BOOL fullscreen)
 		XMoveWindow(xfc->display, window->handle, startX, startY);
 	}
 
-	/* Set the fullscreen state */
-	xf_SendClientEvent(xfc, window->handle, xfc->_NET_WM_STATE, 4,
-	                   fullscreen ? _NET_WM_STATE_ADD : _NET_WM_STATE_REMOVE,
-	                   xfc->_NET_WM_STATE_FULLSCREEN, 0, 0);
+	status = xf_GetWindowProperty(xfc, DefaultRootWindow(xfc->display),
+	                              xfc->_NET_WM_FULLSCREEN_MONITORS, 1, &nitems, &bytes, &prop);
+
+	if (status)
```

Comment:
I was meaning, checking the value of the retrieved property (prop) but perhaps I'm missing something...